src/test/app/forms/models/defendantDetails.ts

```diff
@@ -0,0 +1,205 @@
+/* tslint:disable:no-unused-expression */
+
+import { expect } from 'chai'
+import { Validator } from 'class-validator'
+
+import { expectValidationError } from './validationUtils'
+
+import { DefendantTypes } from 'forms/models/defendantTypes'
+import { DefendantDetails, ValidationErrors } from 'app/forms/models/defendantDetails'
+import * as randomstring from 'randomstring'
+
+describe('Defendant Details', () => {
+  describe('deserialize', () => {
+    it('should return an instance', () => {
+      let deserialized = new DefendantDetails().deserialize({})
+      expect(deserialized).to.be.instanceof(DefendantDetails)
+    })
+
+    it('should return an instance initialised with defaults for "undefined"', () => {
+      expect(new DefendantDetails().deserialize(undefined)).to.eql(new DefendantDetails())
+    })
+
+    it('should return an instance initialised with defaults for "null"', () => {
+      expect(new DefendantDetails().deserialize(null)).to.eql(new DefendantDetails())
+    })
+
+    it('should return am instance from given object', () => {
+      let deserialized = new DefendantDetails().deserialize({
+        type: DefendantTypes.INDIVIDUAL,
+        title: 'title',
+        fullName: 'full name',
+        organisation: undefined,
+        companyHouseNumber: undefined
+      })
+
+      expect(deserialized).to.eql(new DefendantDetails(DefendantTypes.INDIVIDUAL, 'title', 'full name', undefined, undefined))
+    })
+  })
+
+  describe('validation', () => {
+    const validator: Validator = new Validator()
+
+    it('should reject defendant details for undefined type', () => {
+      const errors = validator.validateSync(new DefendantDetails(undefined))
+
+      expect(errors.length).to.equal(1)
+      expectValidationError(errors, ValidationErrors.DEFENDANT_TYPE_REQUIRED)
+    })
+
+    it('should accept defendant details for individual type when individual details are valid', () => {
+      DefendantTypes.all().forEach(type => {
+        const errors = validator.validateSync(new DefendantDetails(DefendantTypes.INDIVIDUAL, 'title', 'full name', undefined, undefined))
+
+        expect(errors.length).to.equal(0)
+      })
+    })
+
+    it('should reject defendant details for individual type when individual details are undefined', () => {
+      DefendantTypes.all().forEach(type => {
+        const errors = validator.validateSync(new DefendantDetails(DefendantTypes.INDIVIDUAL, undefined, undefined, undefined, undefined))
+
+        expect(errors.length).to.equal(2)
```

Comment:
Optional just check for null, so there would be two error as undefined and ''  is considered value. Need to work across model